### DISCUSSION TITLE
gNB does not work on dl_freq

### DISCUSSION THREAD

**Discussion by oguztudo** [srsran/srsRAN_Project#84] [General Help] [answered]
Hello everybody,

i wonder if anybody encountered the same issue but when i run my gNB with my USRP x310 i encounter an issue where my signal is not at the dl_freq. I can only see the Local Oscillator there. However i can see two parts of the Signal at 62.5MHz distance to the left and the right of the CF. Both sides are sending independently. You can see the Waterfall Plot to this here:

![Waterfall](https://github.com/srsran/srsRAN_Project/assets/129198919/f041ecd8-6230-42b4-8003-6bbc0d23ebae)

Here is a closer look on the Spectrum on Max Hold:

![Max_Hold](https://github.com/srsran/srsRAN_Project/assets/129198919/b17c8cbb-9edf-47db-9b47-ef370f1a24b6)

It is funny that when i use the lo_offset, the right part does not move. The left part moves right where the right part is when i set the offset to 62.5 MHz although the distance between both is at 125MHz. This can be seen here:

![Offset](https://github.com/srsran/srsRAN_Project/assets/129198919/ef1b55f1-6aea-413e-a730-6dabc749be0a)

My initial guess was that it is a USRP issue but everything else i did with both of the USRPs i can access worked fine for me so maybe there's an issue in how srsRAN interacts with the USRP? I tried UHD versions 4.1.5 and 4.4.0 and neither worked for me. Here is a screenshot of my started gNB. Further down i added my config (which is pretty basic to be honest) as well as the log file to the gNB. 

![gnb](https://github.com/srsran/srsRAN_Project/assets/129198919/f1d4465b-4fb8-4b03-811a-c532b787ad9f)

[gnb.log](https://github.com/srsran/srsRAN_Project/files/11566149/gnb.log)

Thanks in advance!

**Comment by andrepuschmann** [comment, marked as answer]
On all our X310 we still use UHD 3.15 - I know it shouldn't make a difference but maybe it does.

**Reply by oguztudo** [reply]
Hello Andre,

I agree but it is worth a try. Are you using the official firmware that can be found here:
https://github.com/EttusResearch/uhd/releases/tag/v3.15.0.0

I would really appreciate if you could also give me an Information on the image flavor you are using (HG/XG).

**Reply by andrepuschmann** [reply]
Yes, we've successfully used the official releases. We typically use the XG image to have both SFP ports with 10GigE.

**Reply by andrepuschmann** [reply]
I also suggest you use the UHD examples and verify transmission with your setup works well.

**Reply by oguztudo** [reply]
Hello Andre, sorry i forgot to put an Update:
It indeed worked with UHD 3.15 and also UHD 4.0.0 and I can connect to the gNB using srsUE!
Apparently there is a compatibility issue with newer UHD version.
